CalibTracker/SiStripESProducers edm::Service to ESProducer migration

pieterdavid opened the pull request:
First part of the changes discussed in https://twiki.cern.ch/twiki/bin/view/CMS/SiStrip_edmServicesToESProducersPopConSourceHandlers
Also updated configs using these sources/producers, and moved the `subDetAndLayer` helper method to `TrackerTopology` (as suggested by Gaelle)

pieterdavid committed:
SiStripESProducers edm::Services to ESProducers

pieterdavid committed:
reorganise module selection logic in CalibTracker/SiStripESProducers/plugins/fake/SiStripBadModuleConfigurableFakeESSource.cc

pieterdavid committed:
move subDetAndLayer method to TrackerTopology

cmsbuild commented:
A new Pull Request was created by  @pieterdavid (Pieter David) for CMSSW_8_1_X.

It involves the following packages:

CalibTracker/Configuration
CalibTracker/SiStripESProducers
CalibTracker/SiStripLorentzAngle
DPGAnalysis/SiStripTools
DQM/SiStripCommissioningSources
DQMOffline/Alignment
DataFormats/TrackerCommon


@ghellwig, @monttj, @cvuosalo, @cerminar, @dmitrijus, @cmsbuild, @franzoni, @slava77, @mmusich, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @makortel, @gbenelli, @tocheng, @VinInn, @jlagram, @gpetruc, @rociovilar, @threus, @venturia this is something you requested to watch as well.
@slava77, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 9985f60442a852785eb317b48e6eb7f1036eb50b
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16184/15687/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16184/15687/summary.html

@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * 10021.0_TenMuE_0_200+TenMuE_0_200_pythia8_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017
   * 10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017
   * 10424.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017NewFPix_GenSimFull+DigiFull_2017NewFPix+RecoFull_2017NewFPix+ALCAFull_2017NewFPix+HARVESTFull_2017NewFPix
   * 10624.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017HCALdev_GenSimFull+DigiFull_2017HCALdev+RecoFull_2017HCALdev+ALCAFull_2017HCALdev+HARVESTFull_2017HCALdev
   * 10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017AllNew_GenSimFull+DigiFull_2017AllNew+RecoFull_2017AllNew+ALCAFull_2017AllNew+HARVESTFull_2017AllNew

cvuosalo commented on CalibTracker/Configuration/python/SiStripGain/SiStripGain_Fake_cff.py:
```diff
@@ -1,6 +1,6 @@
 import FWCore.ParameterSet.Config as cms
 
-from CalibTracker.SiStripESProducers.fake.SiStripApvGainFakeESSource_cfi import *    
+from CalibTracker.SiStripESProducers.fake.SiStripApvGainFakeESSource_cfi import *
```
This file seems to have only a change in whitespace. If so, it should not be in this PR.

pieterdavid replied:
Indeed, I reverted it. Thanks for reviewing

cvuosalo commented:
+1

For #16184 9985f60442a852785eb317b48e6eb7f1036eb50b

One step in migrating CalibTracker/SiStripESProducers to multi-threading. Reco code changes are extremely minimal and should not change reco results.

The code changes are satisfactory, and Jenkins tests against baseline CMSSW_8_1_X_2016-10-12-1100 show no significant differences, as expected.

pieterdavid committed:
revert CalibTracker/Configuration/python/SiStripGain/SiStripGain_Fake_cff.py (whitespace changes only)

cmsbuild commented:
Pull request #16184 was updated. @ghellwig, @monttj, @cvuosalo, @cerminar, @dmitrijus, @cmsbuild, @franzoni, @slava77, @mmusich, @vanbesien, @davidlange6 can you please check and sign again.

mmusich commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 514eccc5be4b3f8f04c0243a0b2c3e1585bb0287
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16184/15748/summary.html

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16184/15748/summary.html

cvuosalo commented:
+1

For #16184 514eccc5be4b3f8f04c0243a0b2c3e1585bb0287

Second approval, after removal of file that was not really being changed. New Jenkins tests are still OK.

venturia commented on DataFormats/TrackerCommon/src/TrackerTopology.cc:
```diff
@@ -85,6 +85,29 @@ unsigned int TrackerTopology::module(const DetId &id) const {
   return 0;
 }
 
+std::pair<int,int> TrackerTopology::stripSubDetAndLayer(const DetId& id) const
```
I personally do not like a method called "Strip..." in TrackerTopology. TrackerTopology has no idea whether a subdetector is a SiStrip subdetector or and phase 2 OT subdetector or a pixel subdetector. I see two ways to improve. 
The simple one: a method calle subDetAndLayer in TrackerTopology which returns the subdet (from 1 to 6) and which returns TrackerTopology::layer 
A more complicated one: a method called stripSubDetAndLayer in TrackerGeometry which returns the subdet and the layer ONLY if the detid belongs to a subdetector which, in that geometry is a SiStrip subdetector.

pieterdavid replied:
Good point... actually, after a closer look, I don't think this method belongs in `TrackerTopology` at all (the 'layer' number differs by 1 in with `TrackerTopology::layer`, and ring instead of wheel for TID and TEC) - the numbering scheme it assumes seems to be only used for the parameters passed to a few of the fake ESSources, so I moved it back into CalibTracker/SiStripESProducers (and made a helper class to share the code between the three plugins that use it)

venturia commented:
The documentation in this twiki https://twiki.cern.ch/twiki/bin/view/CMS/SiStripESProducers has to be updated to reflect the present status of the code. For me it is ok if the documentation is moved in the README.md file of the CalibTracker/SiStripESProducers package and a simplified version is left in the twiki

venturia commented:
One more comment: if this configuration DPGAnalysis/SiStripTools/python/SiStripConfObjectAPVPhaseOffsetsFakeESSource_cfi.py is the only one, among the fake source configurations of the SiStrip Tracker, which is not in CalibTracker/SiStripESProducers , I have no problem if it is moved there. Probably it would make more sense if we did so.

pieterdavid committed:
Remove (strip)subDetAndLayer from TrackerTopology

It used a numbering scheme that is only used for the parameters that are
passed to some 'Fake' ESSources. Moved everything related to that into a
small helper class (such that the code is shared between the three
classes that use it).

cmsbuild commented:
Pull request #16184 was updated. @ghellwig, @monttj, @cerminar, @dmitrijus, @cmsbuild, @franzoni, @mmusich, @vanbesien, @davidlange6 can you please check and sign again.

pieterdavid commented:
I think that the configs in SiStripESProducers are mostly examples / test, and the one in DPGAnalysis/SiStripTools logically belongs with some other files there, so the location somehow makes sense, but moving it is easy, please let me know.
I didn't know about the twiki page, I'll have a look and update it

mmusich commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 7ec2ee79a78f7b0b183b39a774962fc91cbeb76e
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16184/15922/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16184/15922/summary.html

boudoul commented:
can the reviewers take a look at this PR ? thanks

boudoul commented:
@ghellwig, @monttj, @cvuosalo, @cerminar, @dmitrijus, @cmsbuild, @franzoni, @slava77, @mmusich, @vanbesien, @davidlange6  : can the reviewers take a look at this PR ? thanks

boudoul commented:
Is there anybody out there  ? 
Song From Pink Floyd album , The Wall

boudoul commented:
![ping](https://cloud.githubusercontent.com/assets/5131720/20055314/bd50ef3a-a4e0-11e6-93b6-77185258fba3.jpg)

boudoul commented:
PING !!
Seriously guys why there is no movement on this PR ?

slava77 commented:
On 11/8/16 9:58 AM, boudoul wrote:
> PING !!
> Seriously guys why there is no movement on this PR ?

DQM and AlCa signatures are needed, based on the default assignments.

Do you expect someone else to also review it?


> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/16184#issuecomment-259081087>, or
> mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEdcbu6dTdo-a5peZAod2qVW46mbBNR8ks5q8DnDgaJpZM4KUgP3>.
>

boudoul commented:
hi @slava77 it's indeed the alca + dqm reviews and eventually signatures that I'm expecting here.

dmitrijus commented:
+1

mmusich commented:
+1
@cmsbuild, please test
@boudoul sorry for long silence, this was never hitting the top of the list 
(test outcomes are long gone, re-triggering them now)

mmusich commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 7ec2ee79a78f7b0b183b39a774962fc91cbeb76e
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16184/16238/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-16184/16238/summary.html
